stats.js

```diff
@@ -22,24 +23,120 @@ const stats = {
 
   updateDistance: async (domain, distance) => {
     const date = new Date().toLocaleDateString();
-    const dayStats = await stats.getDayStats(date);
+    const { dayStats, pausedDomains } = await stats.getDayStats(date);
 
     dayStats.totalDistance += distance;
     dayStats.domains[domain] = {
       distance: (dayStats.domains?.[domain]?.distance || 0) + distance,
     };
 
-    return stats.updateDayStats(date, dayStats);
+    return stats.updateDayStats(date, dayStats, pausedDomains);
+  },
+  // pause tracking on current site
+  pauseCurrentSite: async () => {
+    chrome.tabs.query({ active: true, currentWindow: true }, async (tabs) => {
+      if (tabs.length > 0 && tabs[0].url) {
+        const activeURL = new URL(tabs[0].url);
+        const websiteDomain = activeURL.hostname 
+        const date = new Date().toLocaleDateString();
+        const { dayStats, pausedDomains } = await stats.getDayStats(date);
+        if (!pausedDomains.includes(websiteDomain)) {
+          pausedDomains.push(websiteDomain);
+        }
+        return stats.updateDayStats(date, dayStats, pausedDomains);
+      }
+    });
+  },
+  // check if tracking of current website is paused
+  isPaused: async (domain) => {
+    const date = new Date().toLocaleDateString();
+    const { pausedDomains } = await stats.getDayStats(date);
```

Comment:
Question: are you still saving the `pauseDomains` to today's stats? Don't save it here as we want to keep that state tomorrow.